geode-protobuf/src/main/java/org/apache/geode/internal/protocol/security/processors/AuthorizationSecurityProcessor.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.protocol.security.processors;
+
+import org.apache.geode.internal.protocol.MessageExecutionContext;
+import org.apache.geode.internal.protocol.OperationContext;
+import org.apache.geode.internal.protocol.security.SecurityProcessor;
+
+public class AuthorizationSecurityProcessor implements SecurityProcessor<Object> {
+  @Override
+  public void validateOperation(Object request, MessageExecutionContext messageExecutionContext,
+      OperationContext operationContext) {
+    messageExecutionContext.getAuthorizer().authorize(
```

Comment:
This is going to allow reauthentication (assuming the user is authorized for DATA READ operations), is that what we wanted to do?

Reply:
We should not allow this. Will throw an error

Reply:
The authorizer will process authorization requests. The AuthenticationOperationHandler will block the re-authentication